query/src/main/java/org/infinispan/query/impl/EntityLoader.java

```diff
@@ -44,12 +44,12 @@ public List<E> loadBlocking(List<?> identifiers, Deadline deadline) {
 
       if (queryStatistics.isEnabled()) queryStatistics.entityLoaded(System.nanoTime() - start);
 
-      ArrayList<E> result = new ArrayList<>(entitiesSize);
+      ArrayList<EntityLoaded<E>> result = new ArrayList<>(entitiesSize);
       for (Object key : keys) {
          // if the entity was present at indexing time and
          // it is not present anymore now at searching time,
          // we will add a null here
-         result.add(values.get(key));
+         result.add(new EntityLoaded<>(values.get(key), null));
```

Comment:
Is this an issue with the byte[] get method? And also should we not call getCacheEntries/stream and pass the metadata here as well?

Reply:
@fax4ever I don't think we need `EntityLoader` and `MetadataEntityLoader`. The cost of `getAll` or `getAllCacheEntries` is similar and we can avoid a lot of branches in the code.

Reply:
I would keep these two loading separates. I want to be sure that we don't affect in any way the performance of the queries that do not require metadata, that are 99% of cases.